src/main.c

```diff
@@ -150,7 +151,8 @@ static void do_startup(GObject *object, gpointer data) {
 
 static void do_activate(GObject *object, gpointer data) {
   // because, by default, there are no windows, so the application would quit
-  g_application_hold(G_APPLICATION(app));
+  g_application_hold(G_APPLICATION(object));
+  running = TRUE;
```

Comment:
Can't you just hold the app by default and avoid the `running` bit?

Reply:
The problem is that pressing Ctrl-C seems to send both SIGINT and SIGTERM signals, so g_application_release() is called twice.

Reply:
(part of the problem is due to the trick used to avoid problems with text-only sessions)

Reply:
Ok, I found that I can call g_application_quit several times, so that fixes the problem.